components/gripper/client.go

```diff
@@ -125,3 +108,37 @@ func (c *client) Geometries(ctx context.Context, extra map[string]interface{}) (
 	}
 	return spatialmath.NewGeometriesFromProto(resp.GetGeometries())
 }
+
+func (c *client) Kinematics(ctx context.Context) (referenceframe.Model, error) {
+	// if a model has been cached just return that
+	if c.model != nil {
```

Comment:
I think you are going to need a mutex to protect this struct member.

Reply:
yes thank you